packages/next-server/lib/router/router.ts

```diff
@@ -105,18 +105,18 @@ export default class Router implements BaseRouter {
       return
     }
 
-    // Make sure we don't re-render on initial load,
-    // can be caused by navigating back from an external site
-    if (e.state.options && e.state.options.fromExternal) {
-      return
-    }
-
     // If the downstream application returns falsy, return.
     // They will then be responsible for handling the event.
     if (this._bps && !this._bps(e.state)) {
       return
     }
 
+    // Make sure we don't re-render on initial load,
+    // can be caused by navigating back from an external site
+    if (e.state.options && e.state.options.fromExternal) {
```

Comment:
Why `e.state.options.fromExternal` always return undefined I made sure that the `unload` event is being triggered when visiting an external site but it end up being undefined when navigating back
PS: Doesn't have anything to do with the changes that I made